base/applications/osk/main.c

```diff
@@ -808,10 +826,14 @@ int WINAPI wWinMain(HINSTANCE hInstance,
     /* Load the application's settings from the registry */
     LoadSettings();
 
-    /* If the member of the struct (bShowWarning) is set then display the dialog box */
-    if (Globals.bShowWarning)
+    /* If for whatever reason the thread fails to be created then handle the dialog box in main thread... */
+    if (CreateThread(NULL, 0, (LPTHREAD_START_ROUTINE)OSK_WarningDlgThread, (PVOID)Globals.hInstance, 0, NULL) == NULL)
     {
-        DialogBoxW(Globals.hInstance, MAKEINTRESOURCEW(IDD_WARNINGDIALOG_OSK), Globals.hMainWnd, OSK_WarningProc);
+        /* If the member of the struct (bShowWarning) is set then display the dialog box */
+        if (Globals.bShowWarning)
```

Comment:
You may be able to simplify slightly more all this, and just have one single `if (Globals.bShowWarning)` that if it's true, then you either launch the thread or display the dialog from main thread.
This would also avoid the `bShowWarning` check in the thread implementation.